src/hooks/Circles/usePrivatePosts.ts

```diff
@@ -268,3 +321,131 @@ const createPrivatePostMutationDocument = gql`
     }
   }
 `;
+
+/**
+ * CreatePrivatePostAttachment
+ */
+interface CreatePrivatePostAttachmentMutationProps {
+  userIds: string[];
+  asset: Asset;
+}
+
+export interface CreatePrivatePostAttachmentMutationResult {
+  id: string;
+  fileLocation: {
+    permanentUrl: string;
+    uploadUrl: string;
+  };
+}
+
+const getBlob = (uri: string) => fetch(uri).then((res) => res.blob());
+
+export interface CreateAttachmentResponse {
+  attachment: Attachment;
+  uploadConfig: CreatePrivatePostAttachmentMutationResult;
+}
+
+interface UploadImageProps {
+  asset: Asset;
+  uploadConfig: CreatePrivatePostAttachmentMutationResult;
+}
+
+export function useCreatePrivatePostAttachmentMutation() {
+  const { graphQLClient } = useGraphQLClient();
+  const { data: userData } = useUser();
+
+  const uploadImage = useMutation(['uploadMessageAttachment'], {
+    mutationFn: async ({ asset, uploadConfig }: UploadImageProps) => {
+      if (!asset.uri) {
+        return;
+      }
+
+      const blob = await getBlob(asset.uri);
+      // Use fetch since the s3 uploadUrl is pre-signed
+      await fetch(uploadConfig.fileLocation.uploadUrl, {
+        headers: {
+          'Content-Type': blob.type,
+        },
+        method: 'PUT',
+        body: blob,
+      });
+
+      return uploadConfig;
+    },
+  });
+
+  const createPrivatePostAttachmentMutation = async ({
+    userIds,
+    asset,
+  }: CreatePrivatePostAttachmentMutationProps) => {
+    if (!asset.type) {
+      throw Error("Unknown asset type, can't upload");
+    }
+
+    const fileType = {
+      'image/jpeg': 'JPEG' as const,
+      'image/jpg': 'JPG' as const,
+      'image/png': 'PNG' as const,
+      'image/gif': 'GIF' as const,
+    }[asset.type.toLowerCase()];
+
+    if (!fileType) {
+      throw Error(`Unsupported file type: ${asset.type}`);
+    }
+
+    const variables = {
+      input: {
+        userIds: uniqSort([userData?.id, ...userIds]),
+        fileType,
+      },
+    };
+
+    const res = await graphQLClient.request(
+      createPrivatePostAttachmentMutationDocument,
+      variables,
+    );
+
+    const attachment: Attachment = {
+      externalId: res.privatePostFileUploadUrl.id,
+      subType: 's3',
+      type: {
+        JPEG: AttachmentType.IMAGE,
+        JPG: AttachmentType.IMAGE,
+        PNG: AttachmentType.IMAGE,
+        GIF: AttachmentType.IMAGE,
+      }[fileType],
```

Comment:
Feel like I'm missing something, couldn't this just be

```suggestion
      type: AttachmentType.IMAGE,
```

Reply:
You're not missing anything. This is a relic of an earlier iteration. I originally was trying to add videos too but it had some issues and I wanted to move on and nobody was asking for video uploads.